File: app/src/main/java/com/example/bmicalculator/MainActivity.java
/*
 Łukasz Reinke
 S15037
 Aplikacja mobilna, która realizuje funkcjonalności kalkulatora BMI.
 Dodane następujące testy:
 Analizę statyczną stworzonego kodu. Wynik wyeksportowany do pliku index.html.
 Testy jednostkowe z JUnit.
 Testy Espresso.
*/

package com.example.bmicalculator;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private Button bmiButton, caloriesButton, recipesButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bmiButton = findViewById(R.id.bmiButton);
        caloriesButton = findViewById(R.id.caloriesButton);
        recipesButton = findViewById(R.id.recipesButton);

        bmiButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, BmiActivity.class);
                startActivity(intent);
            }
        });

        caloriesButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, CaloriesActivity.class);
                startActivity(intent);
            }
        });

        recipesButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, RecipesActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Lab4/app/src/main/java/com/example/lab4/MainActivity.kt
package com.example.lab4

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.bmiButton).setOnClickListener {
            startActivity(Intent(this, BmiActivity::class.java))
        }

        findViewById<Button>(R.id.caloriesButton).setOnClickListener {
            startActivity(Intent(this, CaloriesActivity::class.java))
        }

        findViewById<Button>(R.id.recipesButton).setOnClickListener {
            startActivity(Intent(this, RecipesActivity::class.java))
        }
    }
}